site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+## Responsibilities of the Document Object Model (DOM) API
+
+To understand the advantages of a virtual DOM, it is important to understand the role that the DOM itself plays in a web application. The DOM consists of a series of objects, methods, and properties that allow an application to interact with the elements that make up a web page. The DOM `document` is a tree structure that represents the elements that make up a web page. By adding, deleting, and moving elements from the `document`, the appearance and behavior of the page can be altered. Additionally, the DOM contains methods that allow the properties of page elements to be manipulated, providing a way to control the appearance and behavior of individual elements within the page.
+
+For example, if we started with an empty web page, and needed to add an `h1` header containing the page's title, then we could do that with the following statements:
+
+```ts
+  const pageTitle = document.createElement('h1');
+  pageTitle.innerHTML = 'Welcome to Biz-E-Corp!';
+  pageTitle.classList.add('title');
+  document.body.appendChild(pageTitle);
+```
+
+This example starts by creating the `h1` element and then setting two properties - the `innerHTML` property specifies the element's contents and the `classList.add()` method adds a CSS class to the element to alter its appearance. Finally, the element is added to the `document`'s `body` element via a call to that element's `appendChild` method.
+
+The DOM APIs are well-known and reliable, but it can be challenging to use them efficiently. This challenge is caused by the way in which web browsers coordinate changes in the DOM and the page. Whenever a DOM change changes the visible layout of the page, the browser must redetermine the location of each element and redraw them. This means that mundane changes to the DOM can significantly degrade an application's performance.
+
+## Advantages of virtualizing the DOM
+
+> "We can solve any problem by introducing an extra level of indirection" - David J. Wheeler
+
+Ensuring that the DOM is updated efficiently is a critical component to maximizing a web application's performance. Doing that consistently across a large application, however, can be very challenging. Rather than relying on developers to manage this manually, Dojo 2 uses an abstraction layer to manage DOM updates automatically. This abstraction layer is called a *virtual* DOM.
```

Comment:
Rather than relying on developers to manage this manually, Dojo 2 uses an abstraction layer to manage DOM updates automatically. -> Rather than relying on engineering knowledge to remember to always manage DOM operations efficiently, Dojo 2 provides an abstraction layer to manage DOM updates automatically.